companion/gateway/database.py

```
@@ -0,0 +1,535 @@
+    
+    async def get_last_proactive_conversation(self, user_id: str):
+        """Get the last proactive conversation for a user."""
+        query = """
+        SELECT * FROM interactions
+        WHERE user_id = $1 AND is_proactive = true
+        ORDER BY timestamp DESC LIMIT 1
+        """
+        result = await self.execute_user_query(user_id, query, ())
+        return result[0] if result else None
+
+    async def get_recent_interaction_stats(self, user_id: str, days: int = 7):
```

Ruff (0.13.2)

Redefinition of unused `get_recent_interaction_stats` from line 210

(F811)